### PR TITLE
Easier to understand event_dim computation

### LINKED ISSUE #81254: Typo in event_shape of TransformedDistribution?
https://github.com/pytorch/pytorch/blob/0f37d34815fb43db689c3884b49e7dae16a26963/torch/distributions/transformed_distribution.py#L75

I believe there is a bug in the event_shape/batch_shape-"math" of the TransformedDistribution. The line linked above should probably say:
```
        event_dim = transform.codomain.event_dim + max(domain_event_dim - base_event_dim, 0)
```
or rather:
```
        event_dim = transform.codomain.event_dim + max(reinterpreted_batch_ndims, 0)
```

And I can't imagine how `reinterpreted_batch_ndims > 0` would make sense at all. Therefore, I would propose to throw an error in that case. (Then we wouldn't need the `max()` anymore.)

Or am I completely wrong here?  

cc @fritzo @neerajprad @alicanb @nikitaved

### PR DESCRIPTION
**Pull request by johannespitz**
Fixes #81254
Only easier to understand, not a real fix.

### CODE REVIEW

**Comment by kit1980** [comment]
@neerajprad Please take a look if this is something we want to do.

**Comment by neerajprad** [comment]
> @neerajprad Please take a look if this is something we want to do.

This is only for readability, but I find both versions equally readable. :) I will defer to @fritzo on this.

File: torch/distributions/transformed_distribution.py [hunk 1]
```diff
@@ -57,26 +57,29 @@ def __init__(self, base_distribution, transforms, validate_args=None):
...
         transform = ComposeTransform(self.transforms)
-        domain_event_dim = transform.domain.event_dim
-        if len(base_shape) < domain_event_dim:
+        if len(base_shape) < transform.domain.event_dim:
             raise ValueError("base_distribution needs to have shape with size at least {}, but got {}."
-                             .format(domain_event_dim, base_shape))
-        shape = transform.forward_shape(base_shape)
-        expanded_base_shape = transform.inverse_shape(shape)
+                             .format(transform.domain.event_dim, base_shape))
+        sample_shape = transform.forward_shape(base_shape)
```
Comment (fritzo) [inline torch/distributions/transformed_distribution.py:63]: `sample_shape` has precise meaning in `torch.distributions`, and that meaning differs from the proposed use here.  Is there another variable name you could think of using?

**Review by fritzo — CHANGES_REQUESTED** [review]

File: torch/distributions/transformed_distribution.py (hunk 1 above)
Comment (johannespitz) [reply torch/distributions/transformed_distribution.py]: How about `forward_shape`?
I pushed that change, but please feel also free to just close without merging if you feel like it's not worth the trouble.

**Review by fritzo — APPROVED** [review]
LGTM!

**Review by kit1980 — APPROVED** [review]

**Pull request closed without merging**